src/pages/files.js

```diff
@@ -136,18 +143,20 @@ export default class Files extends React.Component {
     });
   };
 
-  handleContinueBacMyStack = () => {
+  getContributionUrl = () => {
     // Get the key url of the file
     const { savedFileUrl } = this.state;
-    const { openCollectiveRedirectUrl } = this.props;
-    const { hostname, protocol } = window.location;
-    const uuid = savedFileUrl.Key.split('/')[0];
-    const jsonUrl = `${protocol}//${hostname}/${uuid}/file/backing.json`;
-    const data = JSON.stringify({
-      jsonUrl,
-    });
-    const contributionUrl = `${openCollectiveRedirectUrl}?data={${data}}`;
-    window.location.replace(contributionUrl);
+    const { openCollectiveRedirectUrl, baseUrl } = this.props;
+    if (savedFileUrl) {
+      const uuid = savedFileUrl.Key.split('/')[0];
+      const jsonUrl = `${baseUrl}/${uuid}/file/backing.json`;
+      const data = JSON.stringify({
+        jsonUrl,
+      });
+      const searchParams = new URLSearchParams(`data=${data}`);
```

Comment:
Why not new `URLSearchParams({data });`?

Reply:
The generated string with that appears in compete, generate something like this
```
%7B%22jsonUrl%22%3A%22http%3A%2F%2Flocalhost%3A3001%2Fb0baf030-a3da-11e9-a530-1ddec81b7392%2Ffile%2Fbacking.json%22%7D%3D

{"jsonUrl":"http://localhost:3001/b0baf030-a3da-11e9-a530-1ddec81b7392/file/backing.json"}=
```

Reply:
Reading the documentation, it might be: `new URLSearchParams(['data', data])`

https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams

Reply:
`new URLSearchParams({ data })`works great for me

Reply:
I was doing something wrong, you're right it works.